core/drivers/cbmem_console.c

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2023, Google Limited
+ */
+
+#include <compiler.h>
+#include <console.h>
+#include <drivers/cbmem_console.h>
+#include <io.h>
+#include <keep.h>
+#include <kernel/dt.h>
+#include <libfdt.h>
+#include <mm/core_mmu.h>
+#include <util.h>
+
+#define CURSOR_MASK (BIT(28) - 1)
+#define OVERFLOW BIT(31)
+
+/*
+ * Structures describing coreboot's in-memory descriptor tables. See
+ * https://github.com/coreboot/coreboot/blob/master/src/commonlib/include/commonlib/coreboot_tables.h
```

Comment:
Prefer a premanent like. Implementation from _blob/master_ may change in the future.

Reply:
I can't find any kind of permanent link (it's documented in the coreboot wiki, but that's being transitioned to a new site...so it's definitely not permanent).  I just used the same language as in the Linux kernel documentation.